contracts/contracts/token/OUSD.sol

```diff
@@ -445,159 +392,267 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
         view
         returns (uint256)
     {
-        if (nonRebasingCreditsPerToken[_account] != 0) {
-            return nonRebasingCreditsPerToken[_account];
+        if (alternativeCreditsPerToken[_account] != 0) {
+            return alternativeCreditsPerToken[_account];
         } else {
-            return _rebasingCreditsPerToken;
+            return rebasingCreditsPerToken_;
         }
     }
 
     /**
-     * @dev Is an account using rebasing accounting or non-rebasing accounting?
-     *      Also, ensure contracts are non-rebasing if they have not opted in.
+     * @dev Auto migrate contracts to be non rebasing,
+     *     unless they have opted into yield.
      * @param _account Address of the account.
      */
-    function _isNonRebasingAccount(address _account) internal returns (bool) {
-        bool isContract = Address.isContract(_account);
-        if (isContract && rebaseState[_account] == RebaseOptions.NotSet) {
-            _ensureRebasingMigration(_account);
+    function _autoMigrate(address _account) internal {
+        bool isContract = _account.code.length > 0;
+        // In previous code versions, contracts would not have had their
+        // rebaseState[_account] set to RebaseOptions.NonRebasing when migrated
+        // therefor we check the actual accounting used on the account instead.
+        if (
+            isContract &&
+            rebaseState[_account] == RebaseOptions.NotSet &&
+            alternativeCreditsPerToken[_account] == 0
+        ) {
+            _rebaseOptOut(_account);
```

Comment:
It should be possible to determine if the code is an AMM by calling the factories to check if it's a known pool. This requires more code integration and could be done via another contract.

Reply:
*nods*. Having it auto optOut gives us 100% compatibility with any contract. We look like a boring ERC20 to anything that doesn't indicate it's willing to deal with the money coming in.